en/middleware/micrortps.md

```diff
@@ -30,58 +32,84 @@ The main elements of the architecture are the client and agent processes shown i
 
 
 ## Code generation
 
-All the code needed to create, build and use the bridge is automatically generated when the PX4 Firmware is compiled. 
+### Code generation for ROS-independent applications
+
+All the code needed to create, build and use the bridge is automatically generated when the PX4 Firmware is compiled.
 
 The *Client* application is also compiled and built into the firmware as part of the normal build process. The *Agent* must be separately/manually compiled for the target computer.
 
 > **Note** [Fast RTPS must be installed](../setup/fast-rtps-installation.md) in order to generate the required code!
 
 <span></span>
-> **Tip** The bridge code can also be [manually generated](micrortps_manual_code_generation.md). Most users will not need to do so, but the linked topic provides a more detailed overview of the build process and can be useful for troubleshooting.
+> **Tip** The bridge code can also be [manually generated](micrortps_manual_code_generation.md). Most users will not need to do
+so, but the linked topic provides a more detailed overview of the build process and can be useful for troubleshooting.
 
+### Code generation for ROS2/ROS applications - `px4_ros_com`
 
-## Supported uORB messages
-
-The generated bridge code will enable a specified subset of uORB topics to be published/subscribed via RTPS.
+The [`px4_ros_com`](https://github.com/PX4/px4_ros_com) package allows the developer to generate all the required ROS2/ROS message headers and sources, but also the agent application which will bridge the client with the RTPS middleware on the ROS2 side. The package is basically composed by two separate branches, where the `master` branch generates all the required ROS2 messages and IDL files, besides some listener and advertiser example nodes, in order to bridge PX4 with ROS2 nodes, and a `ros1` branch, which similarly to the `master` branch, generates the ROS(1) message headers and source files, besides the ROS nodes examples, that allow, by using the `ros1_bridge`, share data between PX4 and ROS.
 
-For *automatic code generation* (via the normal PX4 firmware build process) this set must be listed in the **/path/to/PX4/Firmware/src/modules/micrortps_bridge/CMakeLists.txt** file.
+The `px4_ros_com` package build structure is structured in a way that, when one builds the package, it automatically generates all the required components of the bridge that allow the developer to use the generated messages on its own nodes, being it on the ROS2 or in the ROS side.
```

Comment:
> The `px4_ros_com` package build structure is structured in a way that, when one builds

TO

> The `px4_ros_com` packageis structured such that when one builds the package

Reply:
> it automatically generates all the required components of the bridge that allow the developer to use the generated messages on its own nodes, being it on the ROS2 or in the ROS side."

Again, not clear whether you need this bridge for just ROS1 or also for ROS2. Para here seems to indicate for both. 

This talks about generated messages being used for ROS1?ROS2 nodes, but it isn't clear what components or messages are built with px4_ros_com that a developer might use. Are they the agent messages (from the IDL)?

Reply:
`px4_ros_com` generates:
- on the `master` branch, the ROS2 messages sources and headers, the IDL files required for the `micrortps_agent` and also builds the `micrortps_agent` with the capability to bridge these message topics between the client and ROS2;
- on the `ros1` branch, the ROS(1) messages sources and headers.